src/lib/arch/riscv/InstructionMetadata.hh

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <string>
+
+#include "capstone/capstone.h"
+
+namespace simeng {
+namespace arch {
+namespace riscv {
+
+/** RISCV opcodes. Each opcode represents a unique RISCV operation. */
+namespace Opcode {
+#define GET_INSTRINFO_ENUM
+#include "RISCVGenInstrInfo.inc"
+}  // namespace Opcode
+
+/** A simplified RISCV-only version of the Capstone instruction structure. */
+struct InstructionMetadata {
+ public:
+  /** Constructs a metadata object from a Capstone instruction representation.
+   */
+  InstructionMetadata(const cs_insn& insn);
+
+  /** Constructs an invalid metadata object containing the invalid encoding. */
+  InstructionMetadata(const uint8_t* invalidEncoding, uint8_t bytes = 4);
+
+  static const size_t MAX_OPERAND_STR_LENGTH =
+      sizeof(cs_insn::op_str) / sizeof(char);
+  static const size_t MAX_IMPLICIT_SOURCES =
+      sizeof(cs_detail::regs_read) / sizeof(uint16_t);
+  static const size_t MAX_IMPLICIT_DESTINATIONS =
+      sizeof(cs_detail::regs_write) / sizeof(uint16_t);
+  static const size_t MAX_GROUPS = sizeof(cs_detail::groups) / sizeof(uint8_t);
+  static const size_t MAX_OPERANDS =
+      sizeof(cs_riscv::operands) / sizeof(cs_riscv_op);
```

Comment:
Needs comment per variable to describe what they are

Reply:
This follows the current AArch64 implementation

Reply:
Please make changes in both RISC-V and AArch64 files to add comments